### ISSUE TITLE
RSA encrypt with only public key

### ISSUE THREAD

**Issue by anhpv315** [leocavalcante/encrypt#15]
How can I encrypt a string message with only public key or modulus and exponent?

**Comment by leocavalcante** [comment]
Hm 🤔 
The ideia behind RSA is to be **asymmetric**, a public key for encryption and a private key for decryption.
If you want to use a single key, you should use a **symmetric** system like AES or Salsa20.

**Comment by TheBosZ** [comment]
I think you misunderstood the question.

The question is how to use your library to encrypt a plain text using only the public key.

The `RSA` class requires both the public and private key, which is fine, except that encryption should only require the public key while decryption should only require the private key.

**Comment by leocavalcante** [comment]
Here we go.

`final encrypter = Encrypter(RSA(publicKey: publicKey, privateKey: privateKey));`

If you are just encrypting or just decrypting, you can ignore the respective `privateKey` and `publicKey`. Trying the encrypt without a public key or decrypt without a private key will throw a `StateError`.

This was addressed on v2.0. You can use it right now by referencing it on pub:

```yaml
dependencies:
  kittens:
    git:
      url: git://github.com/leocavalcante/encrypt.git
      ref: v2.0
```

**Comment by TheBosZ** [comment]
This works perfectly! I had to dig a little bit to figure out that I needed to construct an `Encrypted` object using `fromBase64`, but now it's working as I want it to!

Thanks for a useful utility!
